src/bucket/BucketManager.cpp

```diff
@@ -1564,28 +1564,88 @@ BucketManager::mergeBuckets(asio::io_context& ctx,
     return out.getBucket(*this);
 }
 
+template <typename BucketT>
 static bool
-visitLiveEntriesInBucket(
-    std::shared_ptr<LiveBucket const> b, std::string const& name,
-    std::optional<int64_t> minLedger,
-    std::function<bool(LedgerEntry const&)> const& filterEntry,
-    std::function<bool(LedgerEntry const&)> const& acceptEntry,
-    UnorderedSet<Hash>& processedEntries)
+visitBucketEntries(bool visitShadowedEntries, std::shared_ptr<BucketT const> b,
+                   std::string const& name, std::optional<uint32_t> minLedger,
+                   std::function<bool(LedgerEntry const&)> const& filterEntry,
+                   std::function<bool(LedgerEntry const&)> const& acceptEntry,
+                   UnorderedSet<Hash>& processedEntries)
 {
     ZoneScoped;
 
     using namespace std::chrono;
     medida::Timer timer;
 
     bool stopIteration = false;
-    timer.Time([&]() {
-        for (LiveBucketInputIterator in(b); in; ++in)
+
+    auto isLiveEntry = [](auto const& e) {
```

Comment:
This can be replaced with `BucketT::isTombstoneEntry`

Reply:
Done.